autogpt_platform/backend/backend/blocks/http.py

```diff
@@ -68,20 +74,52 @@ def run(self, input_data: Input, **kwargs) -> BlockOutput:
                     # we should send it as plain text instead
                     input_data.json_format = False
 
-        response = requests.request(
-            input_data.method.value,
-            input_data.url,
-            headers=input_data.headers,
-            json=body if input_data.json_format else None,
-            data=body if not input_data.json_format else None,
-        )
-        result = response.json() if input_data.json_format else response.text
-
-        if response.status_code // 100 == 2:
-            yield "response", result
-        elif response.status_code // 100 == 4:
-            yield "client_error", result
-        elif response.status_code // 100 == 5:
-            yield "server_error", result
-        else:
-            raise ValueError(f"Unexpected status code: {response.status_code}")
+        try:
+            response = requests.request(
+                input_data.method.value,
+                input_data.url,
+                headers=input_data.headers,
+                json=body if input_data.json_format else None,
+                data=body if not input_data.json_format else None,
+            )
+            result = response.json() if input_data.json_format else response.text
+
+            if response.ok:
+                yield "response", result
+            elif 400 <= response.status_code < 500:
+                yield "client_error", result
+            elif 500 <= response.status_code < 600:
+                yield "server_error", result
+            else:
+                logger.warning(f"Unexpected status code: {response.status_code}")
+                yield "error", f"Unexpected status code: {response.status_code}"
+
+        except req.exceptions.HTTPError as e:
+            # Handle HTTP errors from raise_for_status()
+            if e.response:
+                status_code = e.response.status_code
+
+                if 400 <= status_code < 500:
+                    result = (
+                        e.response.json()
+                        if input_data.json_format and hasattr(e.response, "json")
```

Comment:
if the `if e.response:` above is removed:
```suggestion
                        if input_data.json_format and e.response
```